wrappers/rust/icicle-core/src/poseidon/mod.rs

```diff
@@ -294,16 +294,23 @@ macro_rules! impl_poseidon {
 #[macro_export]
 macro_rules! impl_poseidon_tests {
     (
-      $field:ident,
-      $field_bytes:literal,
-      $field_prefix:literal,
-      $partial_rounds:literal
+      $field:ident
     ) => {
         #[test]
         fn test_poseidon_hash_many() {
             check_poseidon_hash_many::<$field>()
         }
+    };
+}
 
+#[macro_export]
+macro_rules! impl_poseidon_custom_config_test {
+    (
+      $field:ident,
+      $field_bytes:literal,
+      $field_prefix:literal,
+      $partial_rounds:literal
+    ) => {
```

Comment:
Is there a use case where a curve wouldn't have/need custom config for optimized constants?

Reply:
No, but it is not worth checking in every curve. This test is actually curve independent

Reply:
If there is, maybe we can invoke `impl_poseidon_tests` macro inside this macro to save some copy-paste in curve crates (presumably, if you run `test_poseidon_custom_config` you can always run `test_poseidon_hash_many` right?)

Reply:
>but it is not worth checking in every curve. 

But we are now checking it in every curve?
It also runs the other test inside it as well so we don't need both really

Reply:
tbh, I don't think we even need this custom_config_test at all. And it's really annoying to setup for new curves